PF/Assignment1/ PriorityQueue.java

```diff
@@ -0,0 +1,72 @@
+public class PriorityQueue {
+
+	private TaskNode[] heap;
+	private int heapSize, capacity;
+
+	// Constructor calling
+	public PriorityQueue(int capacity) {
+		this.capacity = capacity + 1;
+		heap = new TaskNode[this.capacity];
+		heapSize = 0;
+	}
+
+	// tocheck whether queue is empty or not
+	public boolean isEmpty() {
+		return heapSize == 0;
+	}
+
+	// checking for heap size
+	public int size() {
+		return heapSize;
+	}
+
+	// insert job in heap based on priority
+	public void insert(String job, int priority) {
+		if (heapSize == capacity - 1) {
+			System.out.println("No more insertion possible");
+			return;
+		}
+
+		TaskNode newJob = new TaskNode(job, priority);
+
+		heap[++heapSize] = newJob;
+		int pos = heapSize;
+		while (pos != 1 && newJob.priority > heap[pos / 2].priority) {
+			heap[pos] = heap[pos / 2];
+			pos = pos / 2;
+		}
+		heap[pos] = newJob;
```

Comment:
ACTION: for first element when heapSize is zero. heap[0] and heap[1] would be allocate with newJob.